### PR TITLE
[PEPPER-843] Fix and test OS1 cohort tag erasure

### PR DESCRIPTION
**Pull request by denniscunningham**
## Context
PEPPER-843

Fixes OS1 cohort tag overwrite by OS2 export. But perhaps the more valuable part of the PR is the testing infra, should we encounter other issues with exported cohort tags, or broader OS1/OS2 export issues.

(Note: I plan to cherry-pick just CohortTagMigrator.java into a separate branch and merge that into staging and production.)

## Checklist

- [x] I have labeled the type of changes involved using the `C-*` labels.
- [x] I have assessed potential risks and labeled using the `R-*` labels.
- [x] I have considered error handling and alerts, and added `L-*` labels as needed.
- [x] I have considered security and privacy, and added `I-*` labels as needed
- [ ] I have analyzed my changes for stability, fault tolerance, graceful degradation, performance bottlenecks and written a brief summary in this PR.
- [ ] If applicable, I have discussed the analytics needs at both a platform and study level with Product and instrumented code accordingly.
- [ ] If applicable, my UI/UX changes have passed muster with Product/Design via an over-the-shoulder review, screenshots, etc.

_If unsure or need help with any of the above items, add the `help wanted` label. For items that starts with `If applicable`, if it is not applicable, check it off and add `n/a` in front._

## FUD Score

_Overall, how are you feeling about these changes?_

- [x] :relaxed: All good, business as usual!
- [ ] :sweat_smile: There might be some issues here
- [ ] :scream: I'm sweaty and nervous

## How do we demo these changes?

_How does one observe these changes in a deployed system? Note that **user visible** encompasses many personas--not just patients and study staff, but also ops duty, your fellow devs, compliance, etc._

- [x] They are user-visible in dev as a regular user journey and require no additional instructions.
- [ ] Getting dev into a state where this is user-visible requires some tech fiddling. I have documented these steps in the related ticket.
- [ ] Requires other features before it's human visible. I have documented the blocking issues in jira.
- [ ] I have no idea how to demo this. Please help me!

## Testing

- [ ] I have added regression test labels to the jira ticket
- [ ] I have added verification steps to the jira ticket
- [x] I have written automated positive tests
- [ ] I have written automated negative tests
- [ ] I have written zero automated tests but have poked around locally to verify proper functionality
- [ ] The jira ticket has acceptance criteria and QA has the needed information to test changes

## Release

- [x] These changes require no special release procedures--just code!
- [ ] Releasing these changes requires special handling and I have documented the special procedures in the release plan document

### CODE REVIEW

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/elastic/migration/CohortTagMigrator.java [hunk 1]
```diff
@@ -32,21 +35,29 @@ private void addOldOsteoTagsToNewOsteo(Map<String, List<CohortTag>> cohortTags,
...
             OldOsteoCohortTagsRetriever oldOsteoRetriever = new OldOsteoCohortTagsRetriever(new CohortTagDaoImpl());
             Map<String, List<CohortTag>> oldOsteoTags = oldOsteoRetriever.retrieve();
+            log.info("Retrieved {} cohort tags via OldOsteoCohortTagsRetriever", oldOsteoTags.size());
             oldOsteoTags.forEach((guid, records) -> {
                 if (cohortTags.containsKey(guid)) {
-                    List<CohortTag> mergedCohorts = Stream.concat(cohortTags.get(guid).stream(), records.stream()).collect(Collectors.toList());
+                    List<CohortTag> mergedCohorts = Stream.concat(cohortTags.get(guid).stream(), records.stream())
+                            .collect(Collectors.toList());
                     cohortTags.put(guid, mergedCohorts);
+                } else {
```
Comment (zyme) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/elastic/migration/CohortTagMigrator.java:44]: Nice, this seems really straightforward!

File: pepper-apis/dsm-core/src/test/java/org/broadinstitute/dsm/model/elastic/migration/CohortTagMigratorTest.java [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+        });
+    }
+
+    @Test
+    public void xxxTest() {
+        // create participants and tags for OS1, OS2 and for both
+        createParticipant(Cohort.OS1);
+        createParticipant(Cohort.OS2);
+        createParticipant(Cohort.OS1_OS2);
+
```
Comment (zyme) [inline pepper-apis/dsm-core/src/test/java/org/broadinstitute/dsm/model/elastic/migration/CohortTagMigratorTest.java:88]: I like this approach--just run the export for different os1/os2 in different ways, and verify that after each export, the tags are what you expect them to be.

**Review by zyme — APPROVED** [review]
Looks great.  I appreciate the quick work here and the thorough automated test!

**Review by ssettipalli — APPROVED** [review]
looks good to me

**Pull request merged**